.github/workflows/preview_release_notes.yml

```diff
@@ -0,0 +1,41 @@
+name: Preview Release Notes
+on:
+  push:
+    branches:
+      - master
+      - release-*
+  pull_request:
+    branches:
+      - master
+      - release-*
+
+jobs:
+  preview_release_notes:
+    name: Preview Release Notes
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: write
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: '0'
+      - name: Setup host
+        uses: ./.github/actions/setup-ubuntu-host
+        with:
+          python-version: '${{ vars.PYTHON_VERSION }}'
+      - name: Generate Release Notes
+        id: generate_release_notes
+        run: python -m scripts.release.release_notes -s $INITIAL_COMMIT_SHA -v $INITIAL_VERSION -o release_notes_tmp.md
+        env:
+          INITIAL_COMMIT_SHA: ${{ vars.RELEASE_INITIAL_COMMIT_SHA }}
+          INITIAL_VERSION: ${{ vars.RELEASE_INITIAL_VERSION }}
+      - name: Summarize results
+        run: cat release_notes_tmp.md >> $GITHUB_STEP_SUMMARY
+      - name: Update PR comment
+        if: github.event_name == 'pull_request'
+        run: gh issue comment $ISSUE --edit-last --create-if-none --body-file release_notes_tmp.md
```

Comment:
Will it work on PRs from forks? I believe in forks the GH token is read only. I use forks and would love to avoid switching to in-repo branches (because it is a mess).

Separate but relevant note - there is `pull_request_target` and `pull_request`  triggers. They have different security considerations. I don't remember the details, but it might be good to look into this.

Reply:
I would probably just make the job required and fail it. Then we can go to the job logs and see what is wrong. Just as with any other test: it doesn't have to post a comment in the PR.

Reply:
Thanks for noticing this. You are actually right, `pull_request` for forks have only read only access token. I've looked into [differences](https://stackoverflow.com/a/74959635) between `pull_request` and `pull_request_target` and GitHub security [blog post](https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/). I think we can go with two options:
- move the PR comment part that needs write access  to `workflow_run`
- if the PR is fork, don't post a comment in PR

We also have a safety valve - we require manual approval via pull request reviews to start GitHub workflows.